TombLib/TombLib/Wad/Catalog/TrCatalog.cs

```diff
@@ -455,10 +456,38 @@ public static string GetVersionString(TRVersion.Game version)
             }
         }
 
-        public static void LoadCatalog(string fileName)
+        private static XmlDocument CombineCatalogs(string rootFolder)
         {
-            XmlDocument document = new XmlDocument();
-            document.Load(fileName);
+			XmlDocument combinedCatalog = new XmlDocument();
+            combinedCatalog.AppendChild(combinedCatalog.CreateXmlDeclaration("1.0", "UTF-8",null));
+            combinedCatalog.AppendChild(combinedCatalog.CreateElement("trcatalog"));
+			foreach (var gameString in TRVersion.NativeVersions.Select(v => v.Native().ToString()))
+			{
+				var gameCatalogFolder = Path.Combine(rootFolder, gameString);
+				if (!Directory.Exists(gameCatalogFolder))
+					throw new InvalidDataException($"The folder {gameCatalogFolder} does not exist for the Game {gameString}");
```

Comment:
I am not sure it should throw exception here. User may intentionally delete some folders, e.g. TEN folder (like they do with .dll files in TRNG). For this case, warning should be more appropriate.